src/Core/Services/MultipleMutationInputValidator.cs

```diff
@@ -0,0 +1,608 @@
+            // current entity (referencing entity) i.e. derivableColumnsFromRequestBody must not contain values for referencing columns.
+            ValidateAbsenceOfReferencingColumnsInTargetEntity(
+                 columnsInTargetEntity: derivableColumnsFromRequestBody,
+                 derivedColumnsFromSourceEntity: columnsDerivedFromSourceEntity,
+                 nestingLevel: nestingLevel,
+                 targetEntityName: entityName,
+                 metadataProvider: metadataProvider);
+
+            // Add all the columns whose value(s) will be derived from insertion in parent entity to the set of derivable columns (b).
+            foreach (string columnDerivedFromSourceEntity in columnsDerivedFromSourceEntity)
+            {
+                derivableColumnsFromRequestBody.TryAdd(columnDerivedFromSourceEntity, parentEntityName);
+            }
+
+            // For the relationships with the parent entity, where the current entity is a referenced entity,
+            // we need to make sure that we have non-null values for all the referenced columns - since the values for all those
+            // columns will be used for insertion in the parent entity. We can get the referenced column value:
+            // Case 1. Via a scalar value provided in the input for the current entity
+            // Case 2. Via another relationship where a column referenced by the parent entity (let's say A) of the current entity (let's say B)
+            // is a referencing column for a target entity (let's say C) related with current entity.
+            // Eg. Suppose there are 3 entities A,B,C which appear in the same order in the multiple mutation.
+            // The relationships defined between the entities are:
+            // A.id (referencing) -> B.id (referenced)
+            // B.id (referencing) -> C.id (referenced)
+            // The value for A.id can be derived from insertion in the table B. B.id in turn can be derived from insertion in the table C.
+            // So, for a mutation like:
+            // mutation {
+            //              createA(item: { aname: "abc", B: { bname: "abc", C: { cname: "abc" } } }) {
+            //                  id
+            //              }
+            //          }
+            // the value for A.id, in effect, will be derived from C.id.
+
+            // Case 1: Remove from columnsToBeDerivedFromEntity, the columns which are autogenerated or
+            // have been provided a non-null value in the input for the current entity.
+
+            // As we iterate through columns which can derived from the current entity, we keep removing them from the
+            // columnsToDerive because we know the column value can be derived.
+            // At the end, if there are still columns yet to be derived, i.e.columnsToDerive.count > 0,
+            // we throw an exception.
+            foreach (string columnToBeDerivedFromEntity in columnsToBeSuppliedToSourceEntity)
```

Comment:
> In the context of processing a GraphQL request, does it make more sense to be consistently saying currentEntity or targetEntity?

I think when we are in this method, everything is in the context of current entity. Parent entity is current entity's parent. Target entity is current entity's child related by some relationship.

>  This inconsistency makes this harder to read because that's even more to keep track of.
Updated/outdated.